libmamba/include/mamba/core/solver.hpp

```diff
@@ -8,6 +8,7 @@
 #define MAMBA_CORE_SOLVER_HPP
 
 #include <string>
+#include <ostream>
```

Comment:
I think `<iosfwd>` would work here, to get only the declarations/forwardings - then you can add `<ostream>` in the cpp implementing the new functions using `std::ostream`